docs/reference/api/connectivity/bluetooth/BLE.md

```diff
@@ -4,6 +4,50 @@ Bluetooth low energy (BLE) is a low power wireless technology standard for build
 
 Arm Mbed BLE, also called `BLE_API`, is the Bluetooth Low Energy software solution for Mbed. Many Mbed [targets and components](https://os.mbed.com/platforms/?mbed-enabled=15&connectivity=3) support Mbed BLE. Developers can use it to create new BLE enabled applications.
 
+Mbed’s `BLE_API` interfaces with the BLE controller on the platform. It hides the BLE stack’s complexity behind C++ abstractions and is compatible with all BLE-enabled Mbed platforms. The Mbed OS `BLE_API` automatically configuring the clocks, timers and other hardware peripherals to work at their lowest power consumption. To do this, yield to the `BLE_API` `waitForEvent()` function whenever the system needs to idle. For more information about `waitForEvent()`, please see our [event-driven programming section](/docs/development/tutorials/the-eventqueue-api.html).
+
+#### `BLE_API`, bridges and stacks
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/BLEDiagram.png)</span>
+
+You can build a BLE application using Mbed OS, `BLE_API` and a controller-specific Bluetooth stack together with some bridge software to adapt it to `BLE_API`:
+
+- `BLE_API` as described above.
+- The bridge software is specific to each vendor’s platform. It provides the instantiations for the interfaces `BLE_API` offers and helps drive the underlying controller and Bluetooth stack.
+- The Bluetooth stack implements the Bluetooth protocol and is specific to the controller, so a vendor using different controllers may provide different stacks.
+
+#### Inside `BLE_API`
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/Inside_API.png)</span>
+
+`BLE_API` offers building blocks to help construct applications. These fall into three broad categories: 
+
+1. Interfaces under **'public/'** to express BLE constructs, such as GAP, GATT, services and characteristics.
+
+2. Code under **'common/'** encapsulates headers that need to be shared between the public interfaces and underlying bridge code.
+
+3. Classes under **'services/'** to offer reference implementations for many of the commonly used GATT profiles. The code under 'services/' isn't essential, but it’s a useful starting point for prototyping. We continue to implement the standard GATT profiles.
+
+#### The BLEDevice class and header
+
+The heart of Mbed's `BLE_API` is the `BLEDevice` class, accessible in the IDE using the header `BLEDevice.h`. This class allows us to create a BLE object that includes the basic attributes of a spec-compatible BLE device and can work with any BLE radio:
```

Comment:
I would rephrase the introduction into: _The entry point of Mbed's `BLE_API` is the BLE class_ accessible using the header `ble/BLE.h`. This class allows us to obtain a BLE object that includes the basic attributes of a spec-compatible BLE device and can work with any BLE radio:_